conans/client/conf/detect.py

```diff
@@ -161,6 +164,55 @@ def _get_profile_compiler_version(compiler, version, output):
     return version
 
 
+def _detect_gcc_libcxx(executable, version, output, profile_name, profile_path):
+    # Assumes a working g++ executable
+    new_abi_available = Version(version) >= Version("5.1")
+    if not new_abi_available:
+        return "libstdc++"
+
+    if not get_env(CONAN_V2_MODE_ENVVAR, False):
+        msg = textwrap.dedent("""
+            Conan detected a GCC version > 5 but has adjusted the 'compiler.libcxx' setting to
+            'libstdc++' for backwards compatibility.
+            Your compiler is likely using the new CXX11 ABI by default (libstdc++11).
+
+            If you want Conan to use the new ABI for the {profile} profile, run:
+
+                $ conan profile update settings.compiler.libcxx=libstdc++11 {profile}
+
+            Or edit '{profile_path}' and set compiler.libcxx=libstdc++11
+            """.format(profile=profile_name, profile_path=profile_path))
+        output.writeln("\n************************* WARNING: GCC OLD ABI COMPATIBILITY "
+                       "***********************\n %s\n************************************"
+                       "************************************************\n\n\n" % msg,
+                       Color.BRIGHT_RED)
+        return "libstdc++"
+
+    main = textwrap.dedent("""
+        #include <string>
+
+        using namespace std;
+        static_assert(sizeof(std::string) != sizeof(void*), "using libstdc++");
+        int main(){}
+        """)
+    t = tempfile.mkdtemp()
+    filename = os.path.join(t, "main.cpp")
+    save(filename, main)
+    old_path = os.getcwd()
+    os.chdir(t)
+    try:
+        error, output = detect_runner("%s main.cpp -std=c++11" % executable)
+        if error:
+            if "using libstdc++" in output:
+                return "libstdc++"
+            # Other error, but can't know, lets keep libstdc++11
+            output.warn("compiler.libcxx check error: %s" % output)
+            output.warn("Couldn't deduce compiler.libcxx for gcc>=5.1, assuming libstdc++11")
+        return "libstdc++11"
```

Comment:
I would add an info message about which libcxx will be used.

Reply:
Agree, will add it to a refactor later